app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -13,18 +13,24 @@ def finish_signup
   end
 
   def finished_signup
-    auth = session['devise.bnet_data']
-    user = User.new(provider: auth['provider'], uid: auth['uid'],
+    @auth = session['devise.bnet_data']
+    user = User.new(provider: @auth['provider'], uid: @auth['uid'],
                     email: params[:email],
-                    battletag: auth['info']['battletag'],
+                    battletag: @auth['info']['battletag'],
                     password: Devise.friendly_token[0,20])
 
     if user.save
+      Player.where(creator_id: User.anonymous, creator_session_id: session.id).
+             update_all(creator_id: user.id)
+      Composition.where(user_id: User.anonymous, session_id: session.id).
+                  update_all(user_id: user.id)
```

Comment:
👍 I'd suggest moving this logic into an `after_create` life cycle hook for `User`, or encapsulate this logic in the `User` model if you're not a fan of AR life cycle hooks. In my mind this is a model concern rather than a controller concern.